File: PRG_410/WEEK_4/Assignment/code_Java/AssignmentWeek4/Rectangle.java

import java.util.*;

/**
 * 
 */
public class Rectangle implements IShape {

    /**
     * Default constructor
     */
    public Rectangle() {
    }

    /**
     * 
     */
    private double mWidth;

    /**
     * 
     */
    private double mHeight;

    /**
     * 
     */
    private void Rectangle() {
        // TODO implement here
    }

    /**
     * @param width 
     * @param height
     */
    public void Rectangle(double width, double height) {
        // TODO implement here
    }

    /**
     * @return
     */
    public double Area() {
        // TODO implement here
        return 0.0d;
    }

    /**
     * @return
     */
    public double Perimeter() {
        // TODO implement here
        return 0.0d;
    }

    /**
     * @return
     */
    public String Type() {
        // TODO implement here
        return "";
    }

    /**
     * @return
     */
    abstract double Area();

    /**
     * @return
     */
    public abstract double Perimeter();

    /**
     * @return
     */
    public abstract String Type();

}

File: PRG_410/WEEK_4/AssignmentWeek4/AssignmentWeek4/Rectangle.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class Rectangle : IShape {

    private Rectangle() {
    }

    private double mWidth;

    private double mHeight;


    /// <summary>
    /// @param width 
    /// @param height
    /// </summary>
    public  Rectangle(double width, double height) {
        mWidth = width;
        mHeight = height;
    }

    /// <summary>
    /// @return
    /// </summary>
    public double Area() {
        // TODO implement here
        return mWidth * mHeight;
    }

    /// <summary>
    /// @return
    /// </summary>
    public double Perimeter() {
        // TODO implement here
        return 2 * (mHeight + mWidth);
    }

    /// <summary>
    /// @return
    /// </summary>
    public String Type() {
        // TODO implement here
        return "Rectangle";
    }

}